### PR TITLE
improve json formatting and composite keys

### PR DESCRIPTION
**Pull request by lennessyy**

### CODE REVIEW

File: modules/API/pages/index.adoc [hunk 1]
```diff
@@ -180,45 +181,47 @@ TYPEDEF TUPLE <field1 INT(1), field2 UINT, field3 STRING(10), field4 DOUBLE> myT
 ----
 
 
-==== Vertices with Composite Keys
+==== Vertices with composite keys
```
Comment (QXG2) [inline modules/API/pages/index.adoc:184]: Please also add such case into "Complex type parameter": https://docs.tigergraph.com/gsql-ref/current/querying/query-operations#_parameter_list_2

Create Query:
CREATE QUERY q_v_person(VERTEX<compositePerson> v) { PRINT v; }
CREATE QUERY q_v_generic(VERTEX v) { PRINT v; }

GSQL Run query:
GSQL > RUN QUERY q_v_person( ("Bob", 123) )
GSQL > RUN QUERY q_v_generic( (("Tom", 456), "compositePerson") )

File: modules/API/pages/built-in-endpoints.adoc [hunk 2]
```diff
@@ -1757,6 +1757,41 @@ For example, the following payload is used to upsert two `User` vertices with an
...
+        "age":{"value":25}
+      },
+      "Tom,456":{
+        "name":{"value":"Tom"},
+        "id":{"value":456},
+        "age":{"value":47}
+      }
+    }
+  }
+}
```
Comment (QXG2) [inline modules/API/pages/built-in-endpoints.adoc:1791]: can you also add the "edges" into this example? like:
```
"edges":{
    "compositePerson":{
      "Bob,123":{
        "compositeRoles":{
          "compositeMovie":{
            "Harry Potter,1990,1337":{
              "role":{
                "value":"Wizard"
              }
            }
          }
        }
      }
    }
  }
```

File: modules/API/pages/index.adoc (hunk 1 above)
Comment (lennessyy) [reply modules/API/pages/index.adoc:184]: yes, i can add this; that is a different repo though and I'll do a separate PR

File: modules/API/pages/built-in-endpoints.adoc (hunk 2 above)
Comment (lennessyy) [reply modules/API/pages/built-in-endpoints.adoc:1806]: added

Comment (QXG2) [reply modules/API/pages/built-in-endpoints.adoc:1806]: Please also add the create vertex/edge for easier understanding.

create vertex compositeMovie (id uint, title string, country string, year uint,  primary key (title,year,id))
create directed edge compositeRoles (from compositePerson,to compositeMovie, role string)

Comment (lennessyy) [reply modules/API/pages/built-in-endpoints.adoc:1808]: added

File: modules/API/pages/built-in-endpoints.adoc [hunk 3]
```diff
@@ -1757,6 +1757,58 @@ For example, the following payload is used to upsert two `User` vertices with an
 }
 ----
 
+==== Composite keys
+
+If your vertex has composite keys, separate the attributes that make up the composite key with comma (`,`) in the same order as they are defined in the schema.
```
Comment (a-m-thomas) [inline modules/API/pages/built-in-endpoints.adoc:1762]:
```suggestion
If your vertex has composite keys, separate the attributes that make up the composite key with a comma (`,`) in the same order as they are defined in the schema.
```

**Review by alexthomasTG — APPROVED** [review]

File: modules/API/pages/built-in-endpoints.adoc [hunk 4]
```diff
@@ -1757,6 +1757,58 @@ For example, the following payload is used to upsert two `User` vertices with an
...
+
+For example, suppose we have the following vertex definition:
+
+[.wrap,gsql]
+----
+CREATE VERTEX Composite_Person(id UINT, name STRING, age UINT, primary key (name, id))
+CREATE DIRECTED EDGE Composite_Roles (from Composite_Person,to Composite_Movie, role string)
```
Comment (QXG2) [inline modules/API/pages/built-in-endpoints.adoc:1769]: Seems this is missing:
```
create vertex compositeMovie (id uint, title string, country string, year uint, primary key (title,year,id))
```

Comment (lennessyy) [reply modules/API/pages/built-in-endpoints.adoc]: added

**Pull request merged**